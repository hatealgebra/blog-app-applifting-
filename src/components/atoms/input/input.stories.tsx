import React from 'react';

import { userEvent, within } from '@storybook/testing-library';

import { expect } from '@storybook/jest';
import { Story } from '@storybook/react';

import { StyledTextArea, StyledTextInput } from './input.styled';

const Template = (args) => <StyledTextInput {...args} />;

export const InputAllSize = () => (
  <div style={{ display: 'flex', flexDirection: 'column', gap: '30px' }}>
    <StyledTextInput placeholder="This is normal input" />
    <StyledTextInput size="big" placeholder="This is big input" />
  </div>
);

export const InputTextVariants = () => (
  <>
    <StyledTextInput type="text" placeholder="Plain text input" />
    <StyledTextInput type="password" placeholder="Plain password input" />
    <StyledTextInput type="email" placeholder="Plain email input" />
    <StyledTextArea rows={1} placeholder="Join the discussion" />
  </>
);

export const FocusedExample = Template.bind({});
FocusedExample.args = {
  placeholder: 'This is a placeholder',
  size: 'big',
};

FocusedExample.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  const input = canvas.getByRole('textbox');
  input.focus();
  await expect(input).toHaveFocus();
};

export const InputWValue = Template.bind({});
InputWValue.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  const input = canvas.getByRole('textbox');
  await userEvent.type(input, 'This is a new comment!');
  await expect(input).toHaveValue('This is a new comment!');
};

export const TextAreaWInput = () => <StyledTextArea />;
TextAreaWInput.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  const input = canvas.getByRole('textbox');
  await userEvent.type(input, 'This is a new comment!');
  await expect(input).toHaveValue('This is a new comment!');
};

export default {
  title: 'Atoms/Input',
  component: StyledTextInput,
  subcomponents: { StyledTextArea },
  decorators: [
    (StoryComponent: Story) => (
      <div style={{ display: 'flex', flexDirection: 'column', gap: '30px' }}>
        <StoryComponent />
      </div>
    ),
  ],
};
