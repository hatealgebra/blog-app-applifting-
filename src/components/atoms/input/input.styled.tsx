import styled, { css } from 'styled-components';

const inputStyling = css`
  width: 100%;
  border: ${({ theme }) => `1px solid ${theme.color.border}`};
  background: white;
  border-radius: 4px;
  font-size: ${({ theme }) => theme.typography.size.h5};
  padding: 0 5px 0 10px;
  &::placeholder {
    color: ${({ theme }) => theme.color.text_muted};
    font-weight: 300;
  }
`;
export const StyledTextInput = styled.input<{ size?: TSize }>`
  ${inputStyling};
  line-height: ${({ size }) => (size === 'big' ? '46px' : '36px')};
`;
export const StyledTextArea = styled.textarea`
  ${inputStyling};
  padding: 10px;
  resize: none;
`;

export type TSize = 'big' | 'normal';
