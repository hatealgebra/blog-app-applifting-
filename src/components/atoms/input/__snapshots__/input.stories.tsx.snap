// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Input InputAllSize smoke-test 1`] = `
<div style="display: flex; flex-direction: column; gap: 30px;">
  <div style="display: flex; flex-direction: column; gap: 30px;">
    <input placeholder="This is normal input"
           class="sc-hLclGa iSKFPv"
    >
    <input placeholder="This is big input"
           class="sc-hLclGa iSKIDA"
    >
  </div>
</div>
`;

exports[`Atoms/Input InputExample smoke-test 1`] = `
<div style="display: flex; flex-direction: column; gap: 30px;">
  <input class="sc-hLclGa iSKFPv">
</div>
`;

exports[`Atoms/Input InputTextVariants smoke-test 1`] = `
<div style="display: flex; flex-direction: column; gap: 30px;">
  <input type="text"
         placeholder="Plain text input"
         class="sc-hLclGa iSKFPv"
  >
  <input type="password"
         placeholder="Plain password input"
         class="sc-hLclGa iSKFPv"
  >
  <input type="email"
         placeholder="Plain email input"
         class="sc-hLclGa iSKFPv"
  >
  <textarea rows="1"
            placeholder="Join the discussion"
            class="sc-eDnVMP fmxlqA"
  >
  </textarea>
</div>
`;
